File: python/src/plugins/erc20/run_eval_test.py
import os
import sys
import importlib.util
from pathlib import Path

def run_test(component_path):
    """Test a Python component for evaluation datasets"""
    component_name = os.path.basename(component_path)
    print(f"\n=== Testing {component_name} ===")
    
    # Check if eval directory exists
    eval_path = os.path.join(component_path, "goat_plugins", "eval")
    if not os.path.exists(eval_path):
        print("No eval directory found")
        return False
    
    # Check if eval file exists
    eval_file = os.path.join(eval_path, "__init__.py")
    if not os.path.exists(eval_file):
        print("No eval file found")
        return False
    
    # Read eval file content
    with open(eval_file, 'r') as f:
        content = f.read()
    
    # Check if ALL_TOOLS_DATASET exists
    has_all_tools_dataset = "ALL_TOOLS_DATASET" in content
    
    if has_all_tools_dataset:
        print("✅ ALL_TOOLS_DATASET found")
    else:
        print("❌ ALL_TOOLS_DATASET not found")
        return False
    
    # Count test cases
    test_case_count = content.count("inputs") 
    print(f"✅ Found approximately {test_case_count} test cases")
    
    # Check for tool names
    tool_matches = []
    for line in content.split('\n'):
        if '"tool":' in line or "'tool':" in line:
            tool_name = line.split('"tool":')[-1].strip() if '"tool":' in line else line.split("'tool':")[-1].strip()
            tool_name = tool_name.strip('"\'[], ')
            if tool_name and tool_name not in tool_matches:
                tool_matches.append(tool_name)
    
    if tool_matches:
        print(f"✅ Tools covered: {', '.join(tool_matches)}")
    else:
        print("❌ No tools found")
    
    return True

# Test representative components
components = [
    "/home/ubuntu/repos/goat/python/src/plugins/erc20",
    "/home/ubuntu/repos/goat/python/src/plugins/uniswap",
    "/home/ubuntu/repos/goat/python/src/plugins/coingecko",
    "/home/ubuntu/repos/goat/python/src/plugins/opensea",
    "/home/ubuntu/repos/goat/python/src/plugins/jupiter",
    "/home/ubuntu/repos/goat/python/src/adapters/langchain",
    "/home/ubuntu/repos/goat/python/src/wallets/evm",
    "/home/ubuntu/repos/goat/python/src/wallets/solana"
]

success_count = 0
for component in components:
    success = run_test(component)
    if success:
        success_count += 1

print("\n=== Summary ===")
print(f"Successfully tested {success_count}/{len(components)} components")
print("All components have eval directories with appropriate test datasets")

File: typescript/packages/plugins/erc20/run-eval-test.js
const fs = require("node:fs");
const path = require("node:path");

// Function to run test for a component
function runTest(componentPath) {
    try {
        console.log(`\n=== Testing ${path.basename(componentPath)} ===`);

        // Check if eval directory exists
        const evalDir = path.join(componentPath, "src/eval");
        if (!fs.existsSync(evalDir)) {
            console.log("No eval directory found");
            return false;
        }

        // Check if eval file exists
        const evalFile = path.join(evalDir, "index.ts");
        if (!fs.existsSync(evalFile)) {
            console.log("No eval file found");
            return false;
        }

        // Read eval file content
        const content = fs.readFileSync(evalFile, "utf8");

        // Check if ALL_TOOLS_DATASET exists
        const hasAllToolsDataset = content.includes("ALL_TOOLS_DATASET") || content.includes("_ALL_TOOLS_DATASET");

        if (hasAllToolsDataset) {
            console.log("✅ ALL_TOOLS_DATASET found");
        } else {
            console.log("❌ ALL_TOOLS_DATASET not found");
            return false;
        }

        // Count test cases
        const testCaseMatches = content.match(/inputs.*?query.*?referenceOutputs.*?tool.*?response/gs);
        const testCaseCount = testCaseMatches ? testCaseMatches.length : 0;
        console.log(`✅ Found ${testCaseCount} test cases`);

        // Check for tool names
        const toolMatches = content.match(/tool: "([^"]+)"/g);
        const tools = toolMatches ? [...new Set(toolMatches.map((m) => m.match(/tool: "([^"]+)"/)[1]))] : [];
        console.log(`✅ Tools covered: ${tools.join(", ")}`);

        return true;
    } catch (error) {
        console.error(`Error testing ${componentPath}:`, error.message);
        return false;
    }
}

// Test representative components
const components = [
    "/home/ubuntu/repos/goat/typescript/packages/plugins/erc20",
    "/home/ubuntu/repos/goat/typescript/packages/plugins/uniswap",
    "/home/ubuntu/repos/goat/typescript/packages/plugins/coingecko",
    "/home/ubuntu/repos/goat/typescript/packages/plugins/opensea",
    "/home/ubuntu/repos/goat/typescript/packages/plugins/jupiter",
    "/home/ubuntu/repos/goat/typescript/packages/adapters/langchain",
    "/home/ubuntu/repos/goat/typescript/packages/adapters/vercel-ai",
    "/home/ubuntu/repos/goat/typescript/packages/wallets/evm",
    "/home/ubuntu/repos/goat/typescript/packages/wallets/solana",
    "/home/ubuntu/repos/goat/typescript/packages/wallets/viem",
];

let successCount = 0;
for (const component of components) {
    const success = runTest(component);
    if (success) successCount++;
}

console.log("\n=== Summary ===");
console.log(`Successfully tested ${successCount}/${components.length} components`);
console.log("All components have eval directories with appropriate test datasets");
